packages/core/src/CSL/cslToCore.ts

```diff
@@ -81,17 +82,158 @@ export const txInputs = (inputs: CSL.TransactionInputs, address?: Cardano.Addres
   return result;
 };
 
-const txCertificates = (_certificates: CSL.Certificates): Cardano.Certificate[] => {
+const stakeRegistration = (certificate: CSL.StakeRegistration): Cardano.StakeAddressCertificate => ({
+  __typename: Cardano.CertificateType.StakeKeyRegistration,
+  rewardAccount: Cardano.RewardAccount(Buffer.from(certificate.to_bytes()).toString())
+});
+
+const stakeDeregistration = (certificate: CSL.StakeDeregistration): Cardano.StakeAddressCertificate => ({
+  __typename: Cardano.CertificateType.StakeKeyDeregistration,
+  rewardAccount: Cardano.RewardAccount(Buffer.from(certificate.to_bytes()).toString())
+});
+
+const stakeDelegation = (certificate: CSL.StakeDelegation): Cardano.StakeDelegationCertificate => ({
+  __typename: Cardano.CertificateType.StakeDelegation,
+  epoch: certificate.epoch(), // TODO: doesn't exist on this type
```

Comment:
That should be nullable for NewTxAlonzo - need to adjust core types. Perhaps just entirely delete it. I'll look at it and let you know.